### ISSUE TITLE
[Pose] Unsatisfactory training results

### ISSUE THREAD

**Issue by Honlan** [NVlabs/few-shot-vid2vid#24]
Hi, thanks a lot for your awesome work.

I try to reproduce the results of pose in the paper. So I 
- download 358 solo dance videos with simple backgrounds
- extract a slice of 30 seconds from each video
- obtain the poses with OpenPose and Densepose for all frames
- run train_g8.sh with 8 Telsa P40

However, after training for 108 epochs, I still cannot reproduce the results and **the face regions are extremely terrible**.

![image](https://user-images.githubusercontent.com/5653648/71790408-4413be80-306b-11ea-9163-03cd64796702.png)

Cound you please give me some advice? 

Another question. The option `n_shot` is set to `1` in `base_option.py`. Should I increase it so that the attention network can be trained during training?

Looking forward to your reply. Thanks again.

**Comment by Yilen-fan** [comment]
Could you provide me your training data ?
I want to train it 
If the results are well , we can share the results , thanks 
my mail : mengmengboy@126.com

**Comment by azizsiyaev** [comment]
Could you please tell how much time did you spend for training with your data?

**Comment by Honlan** [comment]
> Could you please tell how much time did you spend for training with your data?

0.19s per iteration, so 0.19 * 10000 * 100 seconds for the first 100 epochs.

**Comment by howardgriffin** [comment]
I set n_shot to 2 and learning rate to 0.0001, and I got the same result as you. I am also waiting for some advice.

**Comment by howardgriffin** [comment]
I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?

**Comment by Honlan** [comment]
> I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?

I used 8 GPUs to run train_g8.sh

**Comment by howardgriffin** [comment]
> > I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?
> 
> I used 8 GPUs to run train_g8.sh

Did you encounter the problem that as the number of iterations increases, the training stage is slower and slower?

**Comment by Honlan** [comment]
> > > I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?
> > 
> > 
> > I used 8 GPUs to run train_g8.sh
> 
> Did you encounter the problem that as the number of iterations increases, the training stage is slower and slower?

`n_frame_total` will double for every `niter_step` epochs after `niter_single` epochs

**Comment by centosrhel** [comment]
> 
> 
> > > > I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?
> > > 
> > > 
> > > I used 8 GPUs to run train_g8.sh
> > 
> > 
> > Did you encounter the problem that as the number of iterations increases, the training stage is slower and slower?
> 
> `n_frame_total` will double for every `niter_step` epochs after `niter_single` epochs

within `niter_single` epochs, say epoch 39, the training time is tens of that for epoch 1, and the time keeps growing for succeeding epochs. Did you encounter the problem?

**Comment by Honlan** [comment]
> > > > > I used 2 GPUs to run the script and I doubt whether it was due to the multi-GPU?
> > > > 
> > > > 
> > > > I used 8 GPUs to run train_g8.sh
> > > 
> > > 
> > > Did you encounter the problem that as the number of iterations increases, the training stage is slower and slower?
> > 
> > 
> > `n_frame_total` will double for every `niter_step` epochs after `niter_single` epochs
> 
> within `niter_single` epochs, say epoch 39, the training time is tens of that for epoch 1, and the time keeps growing for succeeding epochs. Did you encounter the problem?

I haven't encounter such problem. The training time of a single iteration keeps the same during `niter_single` epochs.

**Comment by howardgriffin** [comment]
I have the same problem as you when training the multi-frame stage and the 'DT_fake' loss is nearly 0. Did you solve it?

**Comment by Honlan** [comment]
> I have the same problem as you when training the multi-frame stage and the 'DT_fake' loss is nearly 0. Did you solve it?

not yet :(
looking forwards to suggestions from the authors

**Comment by AndroYD84** [comment]
I'm experiencing the same issue, [my model started severely artifacting on the face region](https://i.imgur.com/AgLLp3C.jpg) immediately and suddenly at epoch 25, before it was just a blurred face but there were no critical artifacts, I'm using the latest commit from 5 days ago (febb0b1), I started training using all default settings and running [train_g1.sh](https://github.com/NVlabs/few-shot-vid2vid/blob/master/scripts/pose/train_g1.sh) on about 650.000 pictures split in 178 sequences/directories each with a different person.

**Comment by Peng2017** [comment]
This might because some of your openpose-face-keypoints' locations are out of you training images border(256x512). You can try any of the following solutions:
1. different "resize_or_crop" options,
2. carefully clean your dataset images, make sure all faces are located within 256x512 range.
3. Remove you extra-face-discriminator, only use the base generator.
good luck!

**Comment by nihaomiao** [comment]
Hi, @Honlan , could you share the link to these solo dancing videos or give some keywords that I can search online? 

Thanks!

**Comment by tcwang0509** [comment]
This repo is now deprecated. Please refer to the new Imaginaire repo: https://github.com/NVlabs/imaginaire.
